검색 API 구현 및 머지과정에서 필요한 리팩토링

KimJye opened the pull request:
## 작업 내용
검색 API를 작성했습니다.
검색어를 입력하면 영상, 사용자, 태그가 각각 검색이 됩니다.
모두탭에서는 5개씩 보여줍니다.
더보기 또는 영상, 사용자, 태그에서는 검색된 결과에대해 페이지네이션한 결과를 보여줍니다.

users와 같은형태의 파일명과 폴더명을 userList와 같은 형태로 변경하였습니다.
dto에서 배열을 쓰지않기위해서 model로 분리하였습니다.

검색은 인기순으로 정렬됩니다.
영상은 인기도순, 생성순, id순으로 정렬됩니다.
사용자는 비디오를 업로드한 개수 순, 생성된순, id순으로 정렬됩니다.
태그는 태그된 비디오가 많은 순, 생성된순, id순으로 정렬됩니다.

## 실행 방법
검색된 영상의 전체 API: http://localhost:4000/videos?keyword=너무나&page=1
검색된 영상의 개수가 5개일때: http://localhost:4000/videos?keyword=너무나

검색된 사용자의 전체 API: http://localhost:4000/users?keyword=루비&page=1
검색된 사용자의 개수가 5개일때: http://localhost:4000/users?keyword=루비

검색된 태그의 전체 API: http://localhost:4000/tags?keyword=java&page=1
검색된 태그의 개수가 5개일때: http://localhost:4000/tags?keyword=java

KimJye committed:
[server] | feat(user,video,tag): 검색 API 구현 및 리팩토링

검색 API를 구현했습니다.
users와같은 폴더명 또는 파일명을 userList 형태로 모두 변경하였습니다.

KimJye committed:
[server] | refactor(package.json): cross-env start에도 추가

hyokualexkwon commented on packages/server/src/tag/dto/tag-list-request-query.dto.ts:
```diff
@@ -0,0 +1,6 @@
+export class TagListRequestQueryDto {
+  public constructor(
+    public readonly page: string,
```
TagListRequestQueryDto 와 TagListQueryDto 를 한데 묶고, `page: string | number`와 같이 작성해서 공통화하는건 어떨까요?

KimJye replied:
진짜 좋은생각인것같아요 띠요오옹~~ 반영할게요

KimJye committed:
[server] | refactor(): dto 통합으로인한 코드 수정

query string에대한 DTO가 int형 page때문에 2개가 있었습니다.
그래서 하나로 합치고 string | number 형식으로 DTO의 타입을 정했습니다.
그리고 pipe에서는 string으로 받고 int형으로 return 하기 때문에 as
string을 이용하였습니다.

hyokualexkwon commented on packages/server/src/tag/pipe/tag-list-query-pipe.ts:
```diff
@@ -0,0 +1,31 @@
+  }
+
+  private validatePage(page: string): boolean {
+    const onlyNaturalNumberRegex = /^[1-9]\d*$/;
```
정규표현식으로 별도의 상수로 관리하는건 어떨까요?

KimJye replied:
반영했습니다 감사합니다ㅎㅎ

KimJye committed:
[server] | refactor(): 코드 리팩토링

각 모듈에있는 constants를 common의 constants로 옮겼습니다.
pipe에서 page validation에 필요한 정규식을 regexes.ts에 작성하였습니다.

KimJye committed:
[server] | refactor(commentService): 동일한 파일에서 불러온 변수를 하나로 변경

riverandeye commented on packages/server/src/common/constants.ts:
```diff
@@ -9,6 +9,8 @@ export const endpoint = {
   auth: 'auth',
   githubLogin: '/github/login',
   users: 'users',
+  tags: 'tags',
+  videos: 'videos',
```
:+1:

Atercatus commented on packages/server/src/tag/pipe/tag-list-query-pipe.ts:
```diff
@@ -0,0 +1,31 @@
+    return true;
+  }
+
+  private validatePage(page: string): boolean {
```
`isNaturalNumber` 등과 같이 그 역할이 드러나는 네이밍이 좋을 거 같습니다.

KimJye committed:
[server] | refactor(model): model에서 entity라는 이름 삭제

Atercatus commented on packages/server/src/video/pipe/video-list-query-pipe.ts:
```diff
@@ -0,0 +1,63 @@
+    let isValid = true;
+
+    if (
+      (sort === POPULAR && !period) ||
```
이런것도 함수로 빼면 보기에 더 좋을 거 같아요. 다음에 사용해보세연

KimJye committed:
refactor(regexes): 변수명이 더 명확하게 했으면 좋겠다라는 요구사항 반영

KimJye committed:
[server] | refactor(): 상대경로에서 절대경로로 변경

hyokualexkwon reviewed (approved):
👍
